src/azure-cli-core/azure/cli/core/extension/operations.py

```diff
@@ -353,6 +353,15 @@ def add_extension(cmd=None, source=None, extension_name=None, index_url=None, ye
         pass
 
 
+def is_cloud_shell_system_extension(ext_path):
+    import re
+    from azure.cli.core.util import in_cloud_console
+    if in_cloud_console():
+        if re.search(r'^/usr/lib/python3\.\d+/site-packages/azure-cli-extensions/', ext_path):
```

Comment:
Maybe check `startswith` `/usr/lib` should be enough. `re` is slow compared to `startswith`.